django/db/models/functions/datetime.py

```diff
@@ -51,25 +51,31 @@ def __init__(self, expression, lookup_name=None, tzinfo=None, **extra):
         super().__init__(expression, **extra)
 
     def as_sql(self, compiler, connection):
-        if not connection.ops.extract_trunc_lookup_pattern.fullmatch(self.lookup_name):
-            raise ValueError("Invalid lookup_name: %s" % self.lookup_name)
         sql, params = compiler.compile(self.lhs)
         lhs_output_field = self.lhs.output_field
         if isinstance(lhs_output_field, DateTimeField):
             tzname = self.get_tzname()
-            sql = connection.ops.datetime_extract_sql(self.lookup_name, sql, tzname)
+            sql, params = connection.ops.datetime_extract_sql(
+                self.lookup_name, sql, tuple(params), tzname
```

Comment:
is it strictly required to cast params to a tuple? I thought it was understood that `params` may be a list or a tuple throughout the ORM code.

Reply:
This is precisely this inconsistency in the expressions API about `params: list | tuple` that forces the usage of `tuple` here. Some `lhs.as_sql` return `params: list` and but the backend expects layer only expects `tuples` to be provided.

Without this `tuple` some tests crash and I think we want to avoid this expression API inconsistency leak into the backends layer as it could be considered as low level as the compiler which only deals with `params` in tuples.

Reply:
fair enough!